restifizer/RestifizerRequest.m

```diff
@@ -0,0 +1,263 @@
+//
+//  RestifizerRequest.m
+//  Restifizer
+//
+//  Created by Nikita Pankiv on 4/29/16.
+//  Copyright © 2016 VedideV. All rights reserved.
+//
+
+#import "RestifizerRequest.h"
+
+typedef enum : NSUInteger {
+    AuthTypeNone,
+    AuthTypeClient,
+    AuthTypeBearer
+} AuthType;
+
+typedef enum : NSUInteger {
+    RequestTypeGet,
+    RequestTypePost,
+    RequestTypePut,
+    RequestTypePatch,
+    RequestTypeDelete,
+} RequestType;
+
+@interface RestifizerRequest() <NSURLConnectionDataDelegate>
+
+@property(nonatomic, strong) NSString *path;
+@property (nonatomic) AuthType authType;
+@property (nonatomic) RequestType requestType;
+@property (nonatomic) RestifizerAuthParams *authParams;
+@property (nonatomic, strong) RestifizerCompletion completion;
+
+@property (nonatomic) BOOL fetchList;
+@property (nonatomic, strong) NSString *tag;
+
+@property (nonatomic, strong) NSDictionary *parameters;
+@property (nonatomic, strong) NSMutableDictionary *filterParameters;
+@property (nonatomic, strong) NSMutableDictionary *extraQuery;
+
+@property (nonatomic) NSInteger pageNumber;
+@property (nonatomic) NSInteger pageSize;
+
+@property (nonatomic, strong) NSMutableURLRequest *urlRequest;
+@property (nonatomic, strong) NSURLConnection *connection;
+@property (nonatomic) long long contentLength;
+@property (nonatomic, strong) NSMutableData *contentData;
+@property (nonatomic) UInt16 statusCode;
+
+@property (nonatomic, strong) NSDictionary *authTypeNames;
+@end
+
+@implementation RestifizerRequest
+
+- (instancetype)initWithPath:(NSString*)path andAuthParameters:(RestifizerAuthParams*)params {
+    if (self = [super init]) {
+        self.authParams = params;
+        self.path = path;
+        
+        self.authTypeNames = @{@(RequestTypeGet):@"GET",
```

Comment:
@qaz12y, I think it's better to make this object as a static field to avoid memory wasting.